test-math/src/lib.rs

```diff
@@ -27,19 +29,37 @@ mod test_math {
         assert_eq!(pnorm(0.0, 0.0, 1.0, true, false), 0.5);
         assert_eq!(pnorm(0.0, 0.0, 1.0, false, false), 0.5);
         assert_eq!(pnorm(0.0, 0.0, 1.0, false, false), unsafe {
-            c::pnorm5(0.0, 0.0, 1.0, false, false)
+            c::pnorm5(0.0, 0.0, 1.0, 0, 0)
         });
         assert_eq!(pnorm(0.65, 0.2, 0.34, false, false), unsafe {
-            c::pnorm5(0.65, 0.2, 0.34, false, false)
+            c::pnorm5(0.65, 0.2, 0.34, 0, 0)
         });
         assert_eq!(pnorm(3.21, 0.2, 0.34, false, false), unsafe {
-            c::pnorm5(3.21, 0.2, 0.34, false, false)
+            c::pnorm5(3.21, 0.2, 0.34, 0, 0)
         });
         assert_eq!(pnorm(3.21, 0.2, 0.34, false, true), unsafe {
-            c::pnorm5(3.21, 0.2, 0.34, false, true)
+            c::pnorm5(3.21, 0.2, 0.34, 0, 1)
         });
         assert_eq!(pnorm(123.0, 0.2, 0.34, false, true), unsafe {
-            c::pnorm5(123.0, 0.2, 0.34, false, true)
+            c::pnorm5(123.0, 0.2, 0.34, 0, 1)
         });
     }
+
+    #[test]
+    fn test_qnorm() {
+        assert_eq!(qnorm(0.0, 0.5, 1.0, true, false), unsafe {
+            c::qnorm5(0.0, 0.5, 1.0, 1, 0)
+        });
+        assert_eq!(qnorm(0.4, 0.5, 1.0, true, false), unsafe {
+            c::qnorm5(0.4, 0.5, 1.0, 1, 0)
+        });
+        assert_eq!(qnorm(0.4, 0.5, 1.0, false, false), unsafe {
+            c::qnorm5(0.4, 0.5, 1.0, 0, 0)
+        });
+        assert!(abs_diff_eq!(
+            qnorm(-2.3, 0.5, 1.0, false, true),
+            unsafe { c::qnorm5(-2.3, 0.5, 1.0, 0, 1) },
+            epsilon = 1e-15
+        ));
+    }
```

Comment:
Can you add tests to catch the `ML_NAN`s return? Like negative sigma's etc.

Reply:
I had to set `nmath` to `pub` in `src/lib.rs`. Then I could access it via `using rmathlib::nmath::ML_POSINF`.

However, I couldn't access the C macro in Rust. Rust can only access constants and functions from what I could find. See also https://stackoverflow.com/questions/21485655/how-do-i-use-c-preprocessor-macros-with-rusts-ffi

So let's declare checking C macro's as "would have been nice to have", but not easy enough to do? As long as the tests on the top-level functions, such as `qnorm` pass it should be fine.

Reply:
Maybe we can only test

```rust
assert!(result.is_nan())
```
?

Reply:
I see tests as a declarative way to say "given these inputs, I expect this output". However, for constants such as `ML_NAN`, the code itself is already declarative, so adding a test doesn't really add value.